src/main/kotlin/org/jetbrains/intellij/platform/gradle/extensions/IntelliJPlatformDependenciesHelper.kt

```diff
@@ -343,13 +340,69 @@ class IntelliJPlatformDependenciesHelper(
+     * [org.jetbrains.intellij.platform.gradle.extensions.IntelliJPlatformDependenciesHelper.createIntelliJPlatformLocal]
+     * we do not know its location until
+     * [org.jetbrains.intellij.platform.gradle.artifacts.transform.ExtractorTransformer]
+     * has finished. For the local repository we do the same just to keep it simple.
+     */
+    private fun registerIntellijPlatformIvyRepo() {
+        // We can not use IntelliJPlatformRepositoriesHelper.createExclusiveIvyRepository here because it would try
+        // to change already declared repositories and if any of them have been already "used" by something, it would
+        // throw an exception. See:
+        // org.gradle.api.internal.artifacts.repositories.DefaultRepositoryContentDescriptor.assertMutable
+        repositories.forEach {
+            try {
+                it.content {
+                    // For performance reasons exclude the group from already added repos, since we do not expect it to
+                    // exist in any public repositories.
+                    // The ones declared after, should not matter, as long as the artifact is found in this repo,
+                    // because Gradle checks repos in their declaration order.
+                    // Tests on an env with removed caches show that this is actually necessary to prevent extra requests.
+                    excludeGroupAndSubgroups(Dependencies.BUNDLED_MODULE_GROUP)
+                    excludeGroupAndSubgroups(Dependencies.BUNDLED_PLUGIN_GROUP)
+                }
+            } catch (e: Exception) {
+                // Ignore, do not care.
+            }
+        }
+
+        val ivyLocation = if (isIntelliJPlatformLocal()) {
+            // For local IntelliJ platform we can not use pre-create Ivy XML files (see ExtractorTransformer) because it
+            // may be located in location like "/opt/ideaIU" were will not have file system permissions to write.
+            // Because of that we use project's directory.
+            rootProjectDirectory
+        } else {
+            // If not in local mode, then we are using IntelliJ platform processed by the ExtractorTransformer, where
+            // Ivy XML files already exist, and we should add it as a local repository on the fly. This improves
+            // performance because we do not have to locate plugins & create those files, since it is done only once by
+            // the ExtractorTransformer.
+            platformPath.get()
+        }
+
+        // For performance reasons make it exclusive.
+        IntelliJPlatformRepositoriesHelper.createIvyArtifactRepository(
+            "Ivy IntelliJ Platform Artifacts Repository",
+            repositories,
+            providers.localPlatformArtifactsPath(ivyLocation),
+            platformPath.get()
+        ).content {
+            includeGroup(Dependencies.BUNDLED_MODULE_GROUP)
+            includeGroup(Dependencies.BUNDLED_PLUGIN_GROUP)
+        }
+    }
+
```

Comment:
I could create it in `createDefaultRepositores()` as well, so that users have a dedicated method for this, but I thought it would be bad, because then, there we would be updating ivy & artifact path of that repo on the fly. Which is error prone, if this is exposed to customizations.

Reply:
I actually think now, that this approach is better than what was before, because previously (I have not actually tried it) the developer did not call `createDefaultRepositores()` these dependencies won't work, Which is counter intuitive, why we allow defining dependencies which will not work. With this approach if a dependency on a bundled plugin or module is added it will work.